src/ripple/protocol/impl/LedgerFormats.cpp

```diff
@@ -267,6 +267,20 @@ LedgerFormats::LedgerFormats()
             {sfPreviousTxnLgrSeq,    soeREQUIRED}
         },
         commonFields);
+
+    add(jss::AMM,
+        ltAMM,
+        {
+            {sfAccount, soeREQUIRED},
+            {sfTradingFee, soeREQUIRED},
```

Comment:
As noted in an earlier comment, I think `sfTradingFee` can be `soeDEFAULT` to save a little space, as long as any code that sets it calls `makeFieldAbsent` if the new value is 0.